codility_training/lessons.lesson11.SieveofEratosthenes.CountSemiprimes/sAp00n.py

```diff
@@ -1,12 +1,93 @@
+class BST:
+    def __init__(self, data):
+        self.root = Node(data, parent=None)
+
+    def __str__(self):
+        return str(self.in_order_traver(self.root))
+
+    def in_order_traver(self, node):
+        if node.left_child is None:
+            if node.right_child is None:
+                return node.data
+            else:
+                return node.data, self.in_order_traver(node.right_child)
+        else:
+            if node.right_child is None:
+                return self.in_order_traver(node.left_child), node.data
+            else:
+                return self.in_order_traver(node.left_child), node.data, self.in_order_traver(node.right_child)
+
+    def is_in(self, data, node):
+        if node.data == data:
+            # print(f'data already in')
+            return True, node
+        else:
+            if (data > node.data) and (node.right_child is not None):
+                return self.is_in(data, node.right_child)
+            if (data < node.data) and (node.left_child is not None):
+                return self.is_in(data, node.left_child)
+            # print(f'tree don`t have same data\n\n   return_node \n      type: {type(node)}\n        data: {node.data}')
+            return False, node
+
+    def add_new_node(self, data):
+        bul, parent_node = self.is_in(data=data, node=self.root)
+        if not bul:
+            # print('insert new node')
+            new_node = Node(data, parent=parent_node)
+            if new_node.data < parent_node.data:
+                parent_node.left_child = new_node
+            else:
+                parent_node.right_child = new_node
+
+    def compute_number_of_sp(self, start_val, end_val):
+
+        def find_start_node(start_val):
+            node = self.root
+            while not((start_val == node.data) or ((start_val < node.data) and (node.left_child is None))):
+                #print(f'searching_node.data:{node.data}')
+                if node.data > start_val:
+                    node = node.left_child
+                else:
+                    if node.right_child is None:
+                        return None
+                    else:
+                        node = node.right_child
```

Comment:
```suggestion
                    node = node.right_child
```
return 다음에 나오는 else는 필요 없어요~